producer-kafka/src/test/java/io/stargate/producer/kafka/KafkaProducerActivatorTest.java

```diff
@@ -15,268 +15,100 @@
  */
 package io.stargate.producer.kafka;
 
-import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.mockStatic;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import com.codahale.metrics.MetricRegistry;
 import io.stargate.config.store.api.ConfigStore;
-import io.stargate.core.BundleUtils;
 import io.stargate.core.metrics.api.Metrics;
-import io.stargate.db.cdc.CDCProducer;
+import java.util.Hashtable;
+import org.assertj.core.api.Assertions;
 import org.junit.jupiter.api.Test;
-import org.mockito.MockedStatic;
-import org.mockito.Mockito;
 import org.osgi.framework.BundleContext;
+import org.osgi.framework.Filter;
 import org.osgi.framework.InvalidSyntaxException;
-import org.osgi.framework.ServiceEvent;
 import org.osgi.framework.ServiceReference;
 
 class KafkaProducerActivatorTest {
 
   @Test
-  public void shouldNotStartAndRegisterListenersWhenMetricsAndConfigStoreServiceNotProvided()
-      throws InvalidSyntaxException {
+  public void shouldStartIfBothServicesAreRegistered() throws InvalidSyntaxException {
```

Comment:
Is this test critical now that we have dedicated tests for `BaseActivator`?

Reply:
Basic activation should be covered by integrations tests, I assume.

Reply:
only sanity check that it starts when both dependencies are provided. It is not must have but it gives me more confidence that I didn't miss anything.

Reply:
Sure, the full integration test would be better. I don't see any integration test re OSGi in our project, and I didn't investigate ways of writing an IT for OSGi. I can take a look at how it can be implemented.

Reply:
Ah, I assumed we have something under `testing` for Kafka... It's certainly fine to keep this test.